src/middlewared/middlewared/plugins/network.py

```diff
@@ -480,7 +480,9 @@ class DNSService(Service):
         else:
             gc = await self.middleware.call('datastore.query', 'network.globalconfiguration', None, {'get': True})
             if gc['gc_domain']:
-                domain = gc['gc_domain']
+                domains.append('{}\n'.format(gc['gc_domain']))
+            if gc['gc_domains']:
+                domains.append(gc['gc_domains'])
```

Comment:
if you do that you end up with a domains list like: ['domain1', 'extra1\nextra2\nextra3']

You would want ['domain1', 'extra1', 'extra2', 'extra3']